support/camel-k-maven-plugin/src/main/java/org/apache/camel/k/tooling/maven/GenerateCatalogMojo.java

```diff
@@ -524,23 +525,26 @@ private void addCapabilities(RuntimeSpec.Builder runtimeSpec, CamelCatalogSpec.B
         artifacts.clear();
         artifacts.add(Artifact.from("org.apache.camel.quarkus", "camel-quarkus-management"));
         artifacts.add(Artifact.from("org.apache.camel.quarkus", "camel-quarkus-jaxb"));
-        artifacts.add(Artifact.from("org.jolokia", "jolokia-jvm"));
-        addCapabilityAndDependecies(runtimeSpec, catalogSpec, "jolokia", artifacts, false);
+        artifacts.add(Artifact.from("org.jolokia", "jolokia-agent-jvm", "javaagent"));
```

Comment:
According to the test I've done, we require more dependencies. Here the list: https://github.com/apache/camel-k/pull/5090/commits/9072947913a2d3b5360614711b87d8d7df8818c9#diff-96c9016f9dac002d63c2658bc26f05604540fdb0d51d2c87d978747ce0c41a57R235-R265

Reply:
The main artifact is the `jolokia-agent-jvm`, the other dependencies comes transitively from the dependency defined in the camel-k-runtime-bom
The resolved classpath contains the relevant artifacts
```
org.jolokia.jolokia-agent-jvm-2.0.0-javaagent.jar
org.jolokia.jolokia-server-detector-2.0.0.jar
org.jolokia.jolokia-service-discovery-2.0.0.jar
org.jolokia.jolokia-service-history-2.0.0.jar
org.jolokia.jolokia-service-jmx-2.0.0.jar
org.jolokia.jolokia-service-jsr160-2.0.0.jar
org.jolokia.jolokia-service-notif-pull-2.0.0.jar
org.jolokia.jolokia-service-notif-sse-2.0.0.jar
org.jolokia.jolokia-service-serializer-2.0.0.jar
org.apache.camel.camel-management-4.2.0.jar
org.apache.camel.camel-management-api-4.2.0.jar
org.apache.camel.quarkus.camel-quarkus-management-3.6.0.jar
```

Reply:
`jolokia-server-core` seems to be missing. In any case I am not 100% sure of which are the dependencies required as I did some quick try/error experiment. Maybe you can run a test to confirm that nothing else but the agent jar is really required. BTW, since we're there, let's move to 2.0.1.

Reply:
jolokia-server-core is not required, I did a test previously, you can try it too:
```
kamel run <sample.java> -t jolokia.enabled=true
kubectl port-forward $(kubectl get po -l camel.apache.org/integration -oname) 8778:8778
camel hawtio
# add the jmx connection to localhost and jolokia path
```